plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php

```diff
@@ -0,0 +1,412 @@
+<?php
+
+namespace Automattic\WooCommerce\DataBase\Migrations\CustomOrderTable;
+
+use Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController;
+use Automattic\WooCommerce\Internal\DataStores\Orders\DataSynchronizer;
+use WP_CLI;
+
+/**
+ * CLI tool for migrating order data to/from custom table.
+ *
+ * Credits https://github.com/liquidweb/woocommerce-custom-orders-table/blob/develop/includes/class-woocommerce-custom-orders-table-cli.php.
+ *
+ * Class CLIRunner
+ */
+class CLIRunner {
+
+	/**
+	 * CustomOrdersTableController instance.
+	 *
+	 * @var CustomOrdersTableController
+	 */
+	private $controller;
+
+	/**
+	 * DataSynchronizer instance.
+	 *
+	 * @var DataSynchronizer;
+	 */
+	private $synchronizer;
+
+	/**
+	 * PostsToOrdersMigrationController instance.
+	 *
+	 * @var PostsToOrdersMigrationController
+	 */
+	private $post_to_cot_migrator;
+
+	/**
+	 * Init method, invoked by DI container.
+	 *
+	 * @param CustomOrdersTableController      $controller Instance.
+	 * @param DataSynchronizer                 $synchronizer Instance.
+	 * @param PostsToOrdersMigrationController $posts_to_orders_migration_controller Instance.
+	 *
+	 * @internal
+	 */
+	final public function init( CustomOrdersTableController $controller, DataSynchronizer $synchronizer, PostsToOrdersMigrationController $posts_to_orders_migration_controller ) {
+		$this->controller           = $controller;
+		$this->synchronizer         = $synchronizer;
+		$this->post_to_cot_migrator = $posts_to_orders_migration_controller;
+	}
+
+	/**
+	 * Registers commands for CLI.
+	 */
+	public function register_commands() {
+		WP_CLI::add_command( 'wc cot count', array( $this, 'count' ) );
```

Comment:
I may be overthinking this (and I also realize that if I run `wp help wc cot count` everything will be clarified) ... but at face value this name is just a little confusing: it suggests a count of the number of rows in the `wc_orders` table (as opposed to a count of the number of rows in `wp_posts` *still to be migrated).*

Possible alternatives, if we wished to make a change:

- `migration-progress`
- `count-unmigrated`
- `count --unmigrated`

And/or what if the `cot` sub-command was renamed to `cot-migration` or `order-migration`?

Reply:
renamed the `count` method in https://github.com/woocommerce/woocommerce/pull/32938/commits/c36dc7a294dd82a8a5c0046c1bff6312af70409a but have kept the `cot` as same to keep it less verbose. wdyt?

Reply:
Noice :+1: